### PR TITLE
[13.0][MIG] hr_expense_petty_cash

### PR DESCRIPTION
**Pull request by ps-tubtim**
Standard migration.

### CODE REVIEW

File: hr_expense_petty_cash/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# Copyright 2019 Ecosoft Co., Ltd. (http://ecosoft.co.th)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+{
+    "name": "Petty Cash",
+    "version": "13.0.1.0.0",
+    "category": "Human Resources",
+    "author": "Ecosoft, Odoo Community Association (OCA)",
+    "license": "AGPL-3",
+    "website": "https://github.com/OCA/hr",
```
Comment (Saran440) [inline hr_expense_petty_cash/__manifest__.py]:
```suggestion
    "website": "https://github.com/OCA/hr-expense",
```

Comment (ps-tubtim) [reply hr_expense_petty_cash/__manifest__.py]: Thanks for review. Edited.

**Review by Saran440 — COMMENTED** [review]
@ps-tubtim Functional Test.
- check petty cash in vendor bill and save. invoice lines are disappear.
Peek 2020-04-14 09-19

**Comment by ps-tubtim** [comment]
@Saran440 I don't found this bug.
petty_cash

**Comment by Saran440** [comment]
@ps-tubtim okay, it's my mistake.
I config Petty Cash Account is Account Receivable (Allow Reconciliation).

Functional Test:
Step
1. create petty cash limit 100 THB
2. create vendor bills with petty cash 100 THB
3. payment 70 THB
4. create expense 100 THB

I payment 70 THB from 100 THB but I can pay expense more than 70 THB. I'm not sure that it's correct business. How do you think?

**Comment by ps-tubtim** [comment]
@Saran440 
From https://github.com/OCA/hr-expense/pull/6#issuecomment-613248277:
This module computes balance of petty cash holder when vendor bill with petty cash posted. So you can pay expense more than 70 THB but less than 100 THB. And I think this module works correctly according to business process.

**Comment by kittiu** [comment]
@ps-tubtim can you check travis?

**Comment by Saran440** [comment]
please, cherry pick pr https://github.com/OCA/hr/pull/879.

**Review by Saran440 — APPROVED** [review]
Functional Test :+1:

**Comment by newtratip** [comment]
**Test case (pay to petty cash holder)**

1. Create petty cash holder with max limit  1000 THB
2. Create vendor bill with petty cash holder
3. Select invoice line with account petty cash and amount 1500 THB
4. Post invoice
5. Go to Petty Cash menu, I found Balance = 1500 > Max Limit = 1000

I don't sure, Is this correct ?

Peek 2020-10-16 01-29

If I select Petty Cash in the vendor bill, I can not pay to petty cash holder over than max limit amount

Selection_396

File: hr_expense_petty_cash/models/hr_expense_sheet.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        if petty_cash:
+            petty_cash_id = petty_cash_obj.browse(petty_cash)
+            journal = petty_cash_id.journal_id.id or journal
+        return journal
+
+    petty_cash_id = fields.Many2one(
+        string="Petty cash holder",
+        comodel_name="petty.cash",
+        ondelete="restrict",
+        readonly=True,
```
Comment (newtratip) [inline hr_expense_petty_cash/models/hr_expense_sheet.py]:
```suggestion
        
```
readonly=True not necessary for compute field.

**Comment by Saran440** [comment]
@newtratip if you don't check petty cash on vendor,
it mean this bill is not petty cash and not check limit in petty cash.

**Comment by newtratip** [comment]
@Saran440 In your mean, Balance of the petty cash holder can over than max limit amount ?

**Comment by newtratip** [comment]
I have the problem same this https://github.com/OCA/hr-expense/pull/6#issuecomment-613248277

I'm not understand to the business process. If you set the debt vendor bill but you are not payment to the petty cash holder. This is not complete with payment process. Why petty cash holder has amount ?

**Comment by ps-tubtim** [comment]
@Saran440 
I think @newtratip mean we should check the balance of the petty cash holder can't over than max limit amount all case.

**Comment by ps-tubtim** [comment]
@newtratip In the business process, the invoice posted = acknowledge income/expense then the balance of petty cash holder should change after the invoice posted.

**Comment by newtratip** [comment]
> @newtratip In the business process, the invoice posted = acknowledge income/expense then the balance of petty cash holder should change after the invoice posted.

OK

**Review by Siraparn — APPROVED** [review]
Functional Tested

**Comment by kittiu** [comment]
Can this merged?

**Review by newtratip — APPROVED** [review]

**Pull request merged**